app/client/components/delivery/records/deliveryAddressStep.tsx

```diff
@@ -0,0 +1,582 @@
+            address for all of your subscriptions.
+          </p>
+        )}
+        <form
+          action="#"
+          onSubmit={handleFormSubmit(subscriptionNames, productsAffected)}
+        >
+          <fieldset
+            css={css`
+              margin: 0;
+              padding: 0;
+              border: 0;
+              label {
+                margin-top: ${space[3]}px;
+              }
+            `}
+          >
+            <Input
+              label={"Address line 1"}
+              width={30}
+              value={newAddress.addressLine1}
+              changeSetState={(value: string) =>
+                deliveryAddressContext.setAddress?.({
+                  ...newAddress,
+                  addressLine1: value
+                })
+              }
+              inErrorState={
+                status === Status.VALIDATION_ERROR &&
+                !formErrors.addressLine1?.isValid
+              }
+              errorMessage={formErrors.addressLine1?.message}
+            />
+            <Input
+              label="Address line 2"
+              width={30}
+              value={newAddress.addressLine2 || ""}
+              changeSetState={(value: string) =>
+                deliveryAddressContext.setAddress?.({
+                  ...newAddress,
+                  addressLine2: value
+                })
+              }
+              optional={true}
+            />
+            <Input
+              label="Town or City"
+              width={30}
+              value={newAddress.town || ""}
+              changeSetState={(value: string) =>
+                deliveryAddressContext.setAddress?.({
+                  ...newAddress,
+                  town: value
+                })
+              }
+              inErrorState={
+                status === Status.VALIDATION_ERROR && !formErrors.town?.isValid
+              }
+              errorMessage={formErrors.town?.message}
+            />
+            <Input
+              label="County or State"
+              width={30}
+              value={newAddress.region || ""}
+              optional={true}
+              changeSetState={(value: string) =>
+                deliveryAddressContext.setAddress?.({
+                  ...newAddress,
+                  region: value
+                })
+              }
+            />
+            <Input
+              label="Postcode/Zipcode"
+              width={11}
+              value={newAddress.postcode}
+              changeSetState={(value: string) =>
+                deliveryAddressContext.setAddress?.({
+                  ...newAddress,
+                  postcode: value
+                })
+              }
+              inErrorState={
+                status === Status.VALIDATION_ERROR &&
+                !formErrors.postcode?.isValid
+              }
+              errorMessage={formErrors.postcode?.message}
+            />
+            <Select
+              label={"Country"}
+              options={COUNTRIES}
```

Comment:
as discussed IRL we can do a follow-up PR to address the existing country mapping thing (excuse the pun)

Reply:
done - https://github.com/guardian/manage-frontend/pull/396